src/NServiceBus.Persistence.CosmosDB/TransactionalBatchDecoratorExtensions.cs

```diff
@@ -0,0 +1,46 @@
+﻿namespace NServiceBus.Persistence.CosmosDB
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Threading.Tasks;
+
+    static class TransactionalBatchDecoratorExtensions
+    {
+        internal static async Task Execute(this TransactionalBatchDecorator transactionalBatch, Dictionary<int, Operation> operationMappings)
```

Comment:
I'm assuming we'll introduce an overload for a single operation later.

Reply:
Yes, we can add that to the next PR